receiver/jmxreceiver/config_test.go

```diff
@@ -75,6 +76,7 @@ func TestLoadConfig(t *testing.T) {
 				ResourceAttributes: map[string]string{
 					"one": "two",
 				},
+				TempDir: os.TempDir(),
```

Comment:
Please add a test that tests the `temp_dir` field can be set correctly

Reply:
test is here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17884/files#diff-11733262599e85e2533ea74b44796f627ba5c79b13c9e14599d4c4ac9baf44ccR43

Reply:
@atoulme I agree that that test covers the processors ability to use the configuration option correctly, and therefore also tests that the Go field can be set.  What it doesn't test is the ability for a yaml config file to be read in and then correctly map the yaml field to the Go struct's field.  For that we normally use `config_test.go` and the yaml files in `testdata`.

Reply:
Good point! https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17884/commits/4cfc8aec40effe991cfb46c307445d7f3c617286 covers this. Thanks!